### ISSUE TITLE
Several bugs when player quit the game right before server restart.

### ISSUE THREAD

**Issue by k-jiang** [DRE2N/DungeonsXL#407]
When a player quit the game in the middle of a dungeon right before server restart, several bugs occurs when player rejoining the server while there is another new dungeons running:
1. The player still remain inside the DXL_Game_x world. Even though he/she is no longer be considered "inside the dungeon" by DXL, aka the player cannot `/dxl leave` but all other permissions and commands are acting normal like exactly when they are not playing a dungeon.
2. Mobs generated inside all dungeons are being doubled. both [Mob] and [ExternalMob].
3. Other players can [End] their game. But they might not gaining any rewards.

How to reproduce the bugs:
1. Player A starts a new dungeon game.
1. Player A quit the game right before server restart.
1. Restart the server. `timeUntilKickOfflinePlayer` might have an effect here so you could restart the server right before this time.
1. After restarted, let player B start a new dungeon.
1. Let player A join the server while player B is playing.
1. Boom!
We notice that not only player B experiences bugs, other new dungeons created while player A is online also experience the bugs.

Tested with:
DungeonsXL 0.16 build 627
PaperSpigot(A Spigot equivalent) 1.12.2 1428
MythicMobs 4.3.2
Vault 1.5.6

Thank you for your great works on this project! Let me know if you need more information or testing.

**Comment by k-jiang** [comment]
No exceptions are found in the console. If there is only one [Mob] or [ExternalMob] indicated on a single sign, they will not be spawned twice.

**Comment by Sataniel98** [comment]
Thank you for the detailed report, I'll have a look at it!

**Comment by Sataniel98** [comment]
I still could not reproduce this. There might be something big I'm missing, because there have been quite a few bugs recently that I couldn't reproduce and that might be connected to each other.

**Comment by Sataniel98** [comment]
Possibly fixed in build 637.

**Comment by k-jiang** [comment]
Sorry for the late reply.

The first bug (player remaining in a dungeon after server restart) is fixed in build 637.

After a test with build 637, I found out that the mob duplicated problem (the second bug) exists with the normal [Mob] signs. [ExternalMob] seems fine, I have tested it with MythicMobs. In addition, this bug has nothing to do with a server restart. Which means, the mob duplication bug exists all the time.

Off topic: Is it a bug that DXL tries to save an editing dungeon when I `/dxl test` it? Also, how should I quit the test mode and get right back to editing without `/dxl leave` and `/dxl edit xxxx` again?
